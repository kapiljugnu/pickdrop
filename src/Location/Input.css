form label, form input{
    display: block;
    margin: 2px;
}

form input[type="text"]{
    margin-bottom: 30px;
    width: 80%;
}

form input[type="submit"], form input[type="button"]{
    display: inline;
}