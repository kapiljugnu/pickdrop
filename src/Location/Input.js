import React from 'react';
import PropTypes from 'prop-types';
import './Input.css';

const PICK_LOCATION = 'pickLocation';
const DROP_LOCATION = 'dropLocation';

export default class Input extends React.Component {
    static propTypes = {
        onSubmit: PropTypes.func.isRequired,
    };

    state = {
        [PICK_LOCATION]: '',
        [DROP_LOCATION]: '',
    }

    onChange = (field) => (e) => {
        this.setState({ [field]: e.target.value });
    }

    onFormSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit(this.state[PICK_LOCATION], this.state[DROP_LOCATION]);
    }

    onReset = () => {
        this.setState({ [PICK_LOCATION]: '', [DROP_LOCATION]: '' })
    }

    render() {
        return (
            <form>
                <label>Starting Location</label>
                <input type="text" id="pick" value={this.state[PICK_LOCATION]} onChange={this.onChange(PICK_LOCATION)}></input>
                <label>Drop-off point</label>
                <input type="text" id="drop" value={this.state[DROP_LOCATION]} onChange={this.onChange(DROP_LOCATION)}></input>
                <input type="submit" onClick={this.onFormSubmit}></input>
                <input type="button" value="Reset" onClick={this.onReset}></input>
            </form>
        );
    }
}