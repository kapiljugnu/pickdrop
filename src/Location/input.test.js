import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Input from './Input';


describe('Input', () => {
    it('should match the snapshot', () => {
        const inputTree = renderer.create(<Input onSubmit={() => { }} />).toJSON();
        expect(inputTree).toMatchSnapshot();
    })

    it('onSubmit should be called with pick and drop location', () => {
        const onSubmit = jest.fn();
        const LOCATION_PICK = 'location pick';
        const LOCATION_DROP = 'location drop';
        const InputComponent = mount(<Input onSubmit={onSubmit} />);
        const form = InputComponent.find('form');
        form.find('#pick').simulate('change', { target: { value: LOCATION_PICK } });
        form.find('#drop').simulate('change', { target: { value: LOCATION_DROP } });
        form.find('input[type="submit"]').simulate('click');
        expect(onSubmit).toHaveBeenCalledWith(LOCATION_PICK, LOCATION_DROP);
    });

    it('form should be reset on reset click', () => {
        const onSubmit = jest.fn();
        const LOCATION_PICK = 'location pick';
        const LOCATION_DROP = 'location drop';
        const InputComponent = mount(<Input onSubmit={onSubmit} />);
        const form = InputComponent.find('form');
        const pick = form.find('#pick');
        const drop = form.find('#drop');
        pick.simulate('change', { target: { value: LOCATION_PICK } });
        drop.simulate('change', { target: { value: LOCATION_DROP } });
        form.find('input[type="button"]').simulate('click');
        expect(pick.props().value).toBe('');
        expect(drop.props().value).toBe('');
    });
});