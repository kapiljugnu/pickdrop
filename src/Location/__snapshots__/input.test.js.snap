// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match the snapshot 1`] = `
<form>
  <label>
    Starting Location
  </label>
  <input
    id="pick"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label>
    Drop-off point
  </label>
  <input
    id="drop"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    onClick={[Function]}
    type="submit"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Reset"
  />
</form>
`;
